packages/react/src/products/hooks/__tests__/useProductDetails.test.tsx

```diff
@@ -1,75 +1,186 @@
-import { fetchProductDetails } from '@farfetch/blackout-redux';
-import { mockProductId, mockProductsState } from 'tests/__fixtures__/products';
-import { mockStore } from '../../../../tests/helpers';
-import { Provider } from 'react-redux';
-import React from 'react';
-import useProductDetails from '../useProductDetails';
+import {
+  fetchProductDetails,
+  resetProductDetails,
+} from '@farfetch/blackout-redux';
+import {
+  mockProduct,
+  mockProductId,
+  mockProductsState,
+} from 'tests/__fixtures__/products';
 import { renderHook } from '@testing-library/react';
+import { withStore } from '@farfetch/blackout-react/tests/helpers';
+import useProductDetails from '../useProductDetails';
 
 jest.mock('@farfetch/blackout-redux', () => ({
   ...jest.requireActual('@farfetch/blackout-redux'),
-  fetchProductDetails: jest.fn(() => ({ type: 'get' })),
+  fetchProductDetails: jest.fn(() => () => Promise.resolve()),
+  resetProductDetails: jest.fn(() => () => Promise.resolve()),
 }));
 
-const getRenderedHook = (props, state = mockProductsState) => {
-  return renderHook(() => useProductDetails(props), {
-    wrapper: ({ children }) => (
-      <Provider store={mockStore(state)}>{children}</Provider>
-    ),
+describe('useProductDetails', () => {
+  it('should return data correctly with initial state', () => {
+    const { result } = renderHook(() => useProductDetails(mockProductId), {
+      wrapper: withStore(mockProductsState),
+    });
+
+    expect(result.current).toStrictEqual({
+      error: null,
+      isFetched: true,
```

Comment:
I think this should be false. I understand that when `isFetched` is true, we have product data. Is that something that our `isProductFetched` selector is missing?

Reply:
I think the name of this test might be misleading. In this test, the `isFetched` must be true because the store already has product data. 
Instead of `should return data correctly with initial state`, what do you think of `should return data correctly when data is fetched`?

Reply:
My mistake, I commented the wrong line, I will close this and comment the correct line :)